website_event_membership_restriction/views/event_templates.xml

```diff
@@ -0,0 +1,103 @@
+            position="attributes"
+        >
+            <attribute name="class">d-none</attribute>
+        </xpath>
+        <xpath
+            expr="//form//div[@id='o_wevent_tickets_collapse']//button"
+            position="after"
+        >
+            <button
+                type="submit"
+                t-att-class="'btn btn-primary o_wait_lazy_js a-submit ' + (event.allow_members_only and (not request.env.user or request.env.user.partner_id.commercial_partner_id.membership_state not in ['paid', 'free']) and 'disabled' or '')"
+                t-att-disabled="event.allow_members_only and (not request.env.user or request.env.user.partner_id.commercial_partner_id.membership_state not in ['paid', 'free']) and 'disabled' or None"
+                t-attf-id="#{event.id}"
+            >
+                Register
+                <t
+                    t-if="event.seats_limited and event.seats_max and event.seats_available &lt;= (event.seats_max * 0.2)"
+                >
+                    (only <t t-out="event.seats_available" /> available)
+                </t>
+            </button>
+            <t
+                t-if="event.allow_members_only and (not request.env.user or request.env.user.partner_id.commercial_partner_id.membership_state not in ['paid', 'free'])"
+            >
+                <a class="text-center small" href="/membership-info">Members only</a>
+            </t>
+        </xpath>
+        <xpath expr="//form//div[@t-else='']//button" position="attributes">
+            <attribute name="class">d-none</attribute>
+        </xpath>
+        <xpath expr="//form//div[@t-else='']//button" position="after">
+            <button
+                type="submit"
+                t-att-class="'btn btn-primary o_wait_lazy_js a-submit ' + (event.allow_members_only and (not request.env.user or request.env.user.partner_id.commercial_partner_id.membership_state not in ['paid', 'free']) and 'disabled' or '')"
+                t-att-disabled="event.allow_members_only and (not request.env.user or request.env.user.partner_id.commercial_partner_id.membership_state not in ['paid', 'free']) and 'disabled' or None"
+                t-attf-id="#{event.id}"
+            >
+                Register
+                <t
+                    t-if="event.seats_limited and event.seats_max and event.seats_available &lt;= (event.seats_max * 0.2)"
+                >
+                    (only <t t-out="event.seats_available" /> available)
+                </t>
+            </button>
+            <t
+                t-if="event.allow_members_only and (not request.env.user or request.env.user.partner_id.commercial_partner_id.membership_state not in ['paid', 'free'])"
+            >
+                <a class="text-center small" href="/membership-info">Members only</a>
+            </t>
+        </xpath>
+    </template>
+
+    <template id="website_event_membership_info" name="Membership Info">
```

Comment:
This should be a website.page with its `arch`, not a template to be rendered with a controller.